### ISSUE TITLE
Publishing when specifying --version manually still bumps the version

### ISSUE THREAD

**Issue by zabihexes** [deno911/eggs#1]
## 🐛 Bug Report

When I run `eggy publish --version 1.0.11` I would expect that version `1.0.11` gets published.
This is very bad when trying to publish using CI/CD systems.

## To Reproduce

1. Make a nest.land module
2. Try to publish with --version

## Expected behavior

Version `1.0.11` gets published.

## Actual Behavior

Version `1.0.12` gets published.

## Environment

- Eggs version: https://x.nest.land/eggy@0.3.50/cli.ts
- Deno version:
```
deno 1.29.2 (release, x86_64-unknown-linux-gnu)
v8 10.9.194.5
typescript 4.9.4
```
- Operating system and version: devcontainer - `mcr.microsoft.com/vscode/devcontainers/base:debian-10`

**Comment by nberlette** [comment]
@Yaikava This is tied in with the other issue. Pretty sure it's tied in with the CLI using an older version of Cliffy when I picked it up. 

So even though I initially "fixed" it by updating cliffy and some other things, I've discovered it's still using many outdated APIs internally, and they're causing some headaches like this. Sorry about that.

Thank you for taking the time to raise these issues, I'll try to get it taken care of later today!
